client/client.go

```diff
@@ -327,17 +375,16 @@ func (c *client) GetAllMembers(ctx context.Context) ([]*pdpb.Member, error) {
 	defer func() { cmdDurationGetAllMembers.Observe(time.Since(start).Seconds()) }()
 
 	ctx, cancel := context.WithTimeout(ctx, c.timeout)
-	resp, err := c.leaderClient().GetMembers(ctx, &pdpb.GetMembersRequest{
-		Header: c.requestHeader(),
-	})
+	req := &pdpb.GetMembersRequest{Header: c.requestHeader()}
+	ctx = grpcutil.NewReceiverMetadata(ctx, c.GetLeaderAddr())
```

Comment:
If the leader is in network partition, we may not update the members thought leader.